Lerna working builds

nealfennimore opened the pull request:
- Adds in lerna for building between projects
- Abstract out webpack configs for reuse
- Update some [plugins](https://github.com/moderntribe/events-gutenberg/commit/4158517000d6c9857498ab4ed71df8f9fab53cdb)
- Add in webpack and tribe common plugin to global webpack configs
- Add in analyze, dev, and build scripts

mitogh commented on webpack/utils/directories.js:
```diff
@@ -0,0 +1,14 @@
+const { join } = require( 'path' );
+
+const isDirectory = source => lstatSync( source ).isDirectory();
+const getDirectories = source => (
```
Wonder why not inline export instead. `export const ...`

nealfennimore replied:
No this needs to be in `commonjs` syntax - we're not compiling ES6 modules down to CJS syntax before webpack can use it

mitogh replied:
Oh! There was a reason 😆 .

Thanks for the heads up.

mitogh commented on webpack/module/rules/styles.js:
```diff
@@ -0,0 +1,35 @@
+const MiniCssExtractPlugin = require( 'mini-css-extract-plugin' );
+
+const plugin = new MiniCssExtractPlugin( {
```
Same question about the inline export.

mitogh commented on webpack/externals/tribe/common.js:
```diff
@@ -0,0 +1,6 @@
+const pickBy = require( 'lodash/pickBy' );
```
How about `import { pickBy } form 'lodash'`

nealfennimore replied:
Not necessary in this case, lodash is already modularized and this is part of the build process

nealfennimore commented:
@bordoni @mitogh I made a couple changes:
- Updated to use `npm` instead of `yarn`... `yarn` was causing issues with workspaces  and how that was working in conjunction with lerna.
- `@@tribe` is now `@moderntribe`
- `window.tribe.common.hoc` now exists and is properly bundled away from `window.tribe.events`. I now need this file `hoc.js` in the asset pipeline as the events plugin depends on it. 
- `common` plugin has some dependencies on `events` plugin and that can be removed now
- Added in an `analyze` script so you guys can visually see the bundle dependencies (we should probably add a vendor bundle, there's a lot of duplicate vendor modules)

bordoni commented:
I like those changes, my only problem is, do we need to release stuff into NPM? For `@moderntribe`?
